File: Hoimi/Hoimi/Models/Product.cs
namespace Hoimi.Models
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;
    using System.Data.Entity.Spatial;

    [Table("Product")]
    public partial class Product
    {
        [DatabaseGenerated(DatabaseGeneratedOption.None)]
        public int ProductId { get; set; }

        [Required]
        [StringLength(250)]
        public string ProductName { get; set; }

        public int Price { get; set; }
    }
}

File: codes/src/chapter14_refactoring/delivery/v3/Product.java
package chapter14_refactoring.delivery.v3;

class Product {
  final int id;
  final String name;
  final int price;

  Product(final int id, final String name, final int price) {
    this.id = id;
    this.name = name;
    this.price = price;
  }
}
